Extensions/Lighting/lightruntimeobject-pixi-renderer.js

```diff
@@ -391,11 +401,43 @@ gdjs.LightRuntimeObjectPixiRenderer.prototype._computeLightVertices = function (
       obstaclePolygons.push(obstacleHitBoxes[i][j]);
   }
 
+  var maxX = this._object.x + this._radius;
+  var minX = this._object.x - this._radius;
+  var maxY = this._object.y + this._radius;
+  var minY = this._object.y - this._radius;
+
   var flattenVertices = [];
-  for (var i = 0; i < obstaclePolygons.length; i++) {
+  for (var i = 1; i < obstaclePolygons.length; i++) {
     var vertices = obstaclePolygons[i].vertices;
     var verticesCount = vertices.length;
-    for (var j = 0; j < verticesCount; j++) flattenVertices.push(vertices[j]);
+    for (var j = 0; j < verticesCount; j++) {
+      flattenVertices.push(vertices[j]);
+
+      if (vertices[j][0] < minX) minX = vertices[j][0];
+      if (vertices[j][0] > maxX) maxX = vertices[j][0];
+      if (vertices[j][1] < minY) minY = vertices[j][1];
+      if (vertices[j][1] > maxY) maxY = vertices[j][1];
+    }
+  }
+
+  obstaclePolygons[0].vertices[0][0] = minX;
+  obstaclePolygons[0].vertices[0][1] = minY;
+  obstaclePolygons[0].vertices[1][0] = maxX;
+  obstaclePolygons[0].vertices[1][1] = minY;
+  obstaclePolygons[0].vertices[2][0] = maxX;
+  obstaclePolygons[0].vertices[2][1] = maxY;
+  obstaclePolygons[0].vertices[3][0] = minX;
+  obstaclePolygons[0].vertices[3][1] = maxY;
```

Comment:
Is changing the hitbox vertices of light objects like this correct? Or should we create a new `gdjs.Polygon` and use that instead of modifying this?

Reply:
Sorry I'm late to this reply - I think it might be safer to make a duplication of the polygon, otherwise we end up changing the hitbox of the light according to the presence or not of obstacles... would be weird.

Reply:
Note that I'm a bit sad to have to create a new `gdjs.Polygon` every time in this function, but I think it's much much better than modifying the object hitbox. Conceptually, the renderer of an object should not influence the hitbox when rendering it.

Reply:
Actually, we don't need to create a new instance everytime, I guess we can have only one instance and keep modifying it. (Similar to float32 arrays for buffer)

Reply:
Yes totally! We can store this "light bounding polygon" in the light object or pixi runtime renderer :)